.gitignore

```diff
@@ -15,6 +15,7 @@
 /lib/storage/data
 
 # IDE project files
+/.idea
```

Comment:
Please don't ignore these files. They are checked in for a reason.

Reply:
Apologies. I'm using CLion and it was adding a bunch of files which didn't seem necessary, like my shelf files, and a `deployment.xml` and `git_toolbox_prj.xml` files. I didn't realize there were other `.idea` files in the project. At a minimum, I would expect that `.idea/shelf` should be ignored.

Reply:
These seem like specific plugins you have for CLion? If so, please yes gitignore them. Normally the IDE's pretty good about this. Might be because I'm using IntelliJ and/or some folks are using Webstorm, and those plugins might be C++ specific?